zcash_primitives/src/transaction/components/sapling.rs

```diff
@@ -282,12 +353,79 @@ impl SpendDescriptionV5 {
 
 #[derive(Clone)]
 pub struct OutputDescription<Proof> {
-    pub cv: jubjub::ExtendedPoint,
-    pub cmu: bls12_381::Scalar,
-    pub ephemeral_key: EphemeralKeyBytes,
-    pub enc_ciphertext: [u8; 580],
-    pub out_ciphertext: [u8; 80],
-    pub zkproof: Proof,
+    cv: ValueCommitment,
+    cmu: bls12_381::Scalar,
+    ephemeral_key: EphemeralKeyBytes,
+    enc_ciphertext: [u8; 580],
+    out_ciphertext: [u8; 80],
+    zkproof: Proof,
+}
+
+impl<Proof> OutputDescription<Proof> {
+    /// Returns the commitment to the value consumed by this output.
+    pub fn cv(&self) -> &ValueCommitment {
+        &self.cv
+    }
+
+    /// Returns the commitment to the new note being created.
+    pub fn cmu(&self) -> &bls12_381::Scalar {
+        &self.cmu
+    }
+
+    pub fn ephemeral_key(&self) -> &EphemeralKeyBytes {
+        &self.ephemeral_key
+    }
+
+    /// Returns the encrypted note ciphertext.
+    pub fn enc_ciphertext(&self) -> &[u8; 580] {
+        &self.enc_ciphertext
+    }
+
+    /// Returns the output recovery ciphertext.
+    pub fn out_ciphertext(&self) -> &[u8; 80] {
+        &self.out_ciphertext
+    }
+
+    /// Returns the proof for this output.
+    pub fn zkproof(&self) -> &Proof {
+        &self.zkproof
+    }
+}
+
+#[cfg(test)]
+impl<Proof> OutputDescription<Proof> {
+    pub(crate) fn from_parts(
+        cv: ValueCommitment,
+        cmu: bls12_381::Scalar,
+        ephemeral_key: EphemeralKeyBytes,
+        enc_ciphertext: [u8; 580],
+        out_ciphertext: [u8; 80],
+        zkproof: Proof,
+    ) -> Self {
+        OutputDescription {
+            cv,
+            cmu,
+            ephemeral_key,
+            enc_ciphertext,
+            out_ciphertext,
+            zkproof,
+        }
+    }
+    pub(crate) fn cv_mut(&mut self) -> &mut ValueCommitment {
+        &mut self.cv
+    }
+    pub(crate) fn cmu_mut(&mut self) -> &mut bls12_381::Scalar {
+        &mut self.cmu
+    }
+    pub(crate) fn ephemeral_key_mut(&mut self) -> &mut EphemeralKeyBytes {
+        &mut self.ephemeral_key
+    }
+    pub(crate) fn enc_ciphertext_mut(&mut self) -> &mut [u8; 580] {
+        &mut self.enc_ciphertext
+    }
+    pub(crate) fn out_ciphertext_mut(&mut self) -> &mut [u8; 80] {
```

Comment:
Zingolib's test framework also uses these. Could we
```suggestion
#[cfg(any(test, feature = "test-dependencies"))]
impl<Proof> OutputDescription<Proof> {
    pub fn from_parts(
        cv: ValueCommitment,
        cmu: bls12_381::Scalar,
        ephemeral_key: EphemeralKeyBytes,
        enc_ciphertext: [u8; 580],
        out_ciphertext: [u8; 80],
        zkproof: Proof,
    ) -> Self {
        OutputDescription {
            cv,
            cmu,
            ephemeral_key,
            enc_ciphertext,
            out_ciphertext,
            zkproof,
        }
    }
    pub fn cv_mut(&mut self) -> &mut ValueCommitment {
        &mut self.cv
    }
    pub fn cmu_mut(&mut self) -> &mut bls12_381::Scalar {
        &mut self.cmu
    }
    pub fn ephemeral_key_mut(&mut self) -> &mut EphemeralKeyBytes {
        &mut self.ephemeral_key
    }
    pub fn enc_ciphertext_mut(&mut self) -> &mut [u8; 580] {
        &mut self.enc_ciphertext
    }
    pub fn out_ciphertext_mut(&mut self) -> &mut [u8; 80] {
```

Reply:
We could add them behind the `test-dependencies` feature instead, but I think that they're better off only for internal testing. Even better would be if we could exclude the need for any mutation from our tests.